pkg/clusteragent/admission/mutate/cwsinstrumentation/cws_instrumentation.go

```diff
@@ -249,6 +308,23 @@ func NewCWSInstrumentation() (*CWSInstrumentation, error) {
 		ci.image = fmt.Sprintf("%s/%s", cwsInjectorContainerRegistry, ci.image)
 	}
 
+	// parse mode
+	ci.mode, err = ParseInstrumentationMode(config.Datadog.GetString("admission_controller.cws_instrumentation.mode"))
+	if err != nil {
+		return nil, fmt.Errorf("can't initiatilize CWS Instrumentation: %v", err)
+	}
+	ci.mountVolumeForRemoteCopy = config.Datadog.GetBool("admission_controller.cws_instrumentation.remote_copy.mount_volume")
+
+	if ci.mode == RemoteCopy {
+		// build the cluster agent service account
+		serviceAccountName := config.Datadog.GetString("cluster_agent.service_account_name")
+		if len(serviceAccountName) == 0 {
+			return nil, fmt.Errorf("can't initialize CWS Instrumentation in %s mode without providing a service account name in config (cluster_agent.service_account_name)", RemoteCopy)
+		}
+		ns := config.Datadog.GetString("kube_resources_namespace")
```

Comment:
You should use: https://github.com/DataDog/datadog-agent/blob/main/pkg/util/kubernetes/apiserver/common/common.go#L40-L49

I was also thinking about the multiple Cluster Agent deployments in same cluster. Normally it does not happen, but mentioning just in case (the command filtering should match)

Reply:
Thanks for the pointer ! I pushed a commit to use the fonction right above the one you suggested so that it leverages both the env variable (particularly useful for tests 😅) and the file.

As for the multiple cluster agent issue, that's a valid point - as you said the command filtering should match ... I'll try to think of other filtering solutions for a v2 !

Reply:
I should have clarified, you need to use `GetMyNamespace` and not `GetResourcesNamespace` as it's possible that `kube_resources_namespace` is not Cluster Agent POD namespace, while it's impossible to reference a service account from another namespace in a POD.

Reply:
Ok, I'll change ... but it means the test will be less accurate 😭